src/dapp/src/features/vote/VoteDetails.tsx

```diff
@@ -1,52 +1,82 @@
+import { useEffect, useState } from 'react'
 import { VotingRoundPopulated } from '../../shared/types'
-import { getWalletLabel } from '../../shared/wallet'
+import { NFDomain, fetchNFDomain, getWalletLabel } from '../../shared/wallet'
 import { VotingTime } from './VotingTime'
 
 type VoteDetailsProps = {
   loading: boolean
   round: VotingRoundPopulated | undefined | null
 }
-export const VoteDetails = ({ loading, round }: VoteDetailsProps) => (
-  <>
-    {round && (
-      <Stack spacing={1}>
-        <Typography className="mt-5 font-bold sm:font-normal" variant="h5">
-          Vote details
-        </Typography>
-        <VotingTime className="hidden sm:block mt-4" loading={loading} round={round} />
-        {loading ? (
-          <Skeleton variant="text" />
-        ) : (
-          <Typography>
-            Voting round created by{' '}
-            <Link href={`${import.meta.env.VITE_ALGO_EXPLORER_URL}/address/${round.created.by}`} target="_blank" className="font-normal">
-              {getWalletLabel(round.created.by)}
-            </Link>
+export const VoteDetails = ({ loading, round }: VoteDetailsProps) => {
+  const [nfDomain, setNfDomain] = useState<NFDomain | undefined>(undefined)
+  const [isFetchingNfDomain, setIsFetchingNfDomain] = useState(false)
+
+  useEffect(() => {
+    if (round) {
+      setIsFetchingNfDomain(true)
+      fetchNFDomain(round.created.by).then((data) => {
+        setNfDomain(data)
+        setIsFetchingNfDomain(false)
+      })
+    }
+  }, [round])
+
+  return (
+    <>
+      {round && (
+        <Stack spacing={1}>
+          <Typography className="mt-5 font-bold sm:font-normal" variant="h5">
+            Vote details
           </Typography>
-        )}
-        {loading ? (
-          <Skeleton variant="text" />
-        ) : (
-          <Link href={`${import.meta.env.VITE_ALGO_EXPLORER_URL}/application/${round.id}`} target="_blank">
-            Smart contract
-          </Link>
-        )}
-        {loading ? (
-          <Skeleton variant="text" />
-        ) : (
-          <Link href={`${import.meta.env.VITE_IPFS_GATEWAY_URL}/${round.cid}`} target="_blank">
-            Voting round details in IPFS
-          </Link>
-        )}
-        {loading ? (
-          <Skeleton variant="text" />
-        ) : round.snapshot?.snapshot.length ? (
-          <Link href={`${import.meta.env.VITE_IPFS_GATEWAY_URL}/${round.voteGatingSnapshotCid}`} target="_blank">
-            Allow list
-          </Link>
-        ) : null}
-      </Stack>
-    )}
-  </>
-)
+          <VotingTime className="hidden sm:block mt-4" loading={loading} round={round} />
+          {loading || isFetchingNfDomain ? (
+            <Skeleton variant="text" />
+          ) : (
+            <Typography>
+              Voting round created by{' '}
+              {nfDomain && nfDomain.name ? (
+                <Link
+                  href={`https://app${import.meta.env.VITE_IS_TESTNET ? '.testnet' : ''}.nf.domains/name/${nfDomain.name}`}
```

Comment:
Minor: to avoid polluting this file you could create a component that displays an account, it would show the raw account while loading and then switch it out with the NF Domain. No need for skeleton and easier to reuse across the site.